### PR TITLE
[DF][RCsvDS] Allow user-specified columns; Handle NaN values

### LINKED ISSUE #9561: [DF] Teach CSV datasource to handle NaN values
Original report, it also contains a test: https://root-forum.cern.ch/t/nan-values-in-one-column-break-histogram-of-another-column/37091 .

### PR DESCRIPTION
**Pull request by ikabadzhov**
Users can now pass a map to the `RCsvDS` constructor as an additional
argument to specify column types of their choice.
Supported types remain: double, Long64_t, bool, std::string.
Type aliases follow `TTreeReader`, i.e.:
* 'O' for boolean
* 'D' for double
* 'L' for Long64_t
* 'T' for std::string

`RCsvDS` now supports ready empty cells and now introduces special variables "NaN" and "nan" to be read as `nan`. Note that only double has actual `nan`, so for booleans and Long64_t-s, 0 is set in this case.

Corresponding tests are added.

- [x] tested changes locally
- [x] updated the docs (if necessary)

This PR fixes #9561

### CODE REVIEW

File: tree/dataframe/test/datasource_csv.cxx [hunk 1]
```diff
@@ -321,4 +323,69 @@ TEST(RCsvDS, ProgressiveReadingRDFMT)
...
+   const std::string Warn =
+      "Column \"col3\" of type bool contains empty cell(s) or NaN(s).\n"
+      "There is no `nan` equivalent for type bool, hence `false` is stored.\n"
+      "Column \"col5\" of type Long64_t contains empty cell(s) or NaN(s).\n"
+      "There is no `nan` equivalent for type Long64_t, hence `0` is stored.\n"
+      "Column \"col6\" of type Long64_t contains empty cell(s) or NaN(s).\n"
+      "There is no `nan` equivalent for type Long64_t, hence `0` is stored.\n"
+      "Please manually set the column type to `double` (with `D`) in `MakeCsvDataFrame` to read NaNs instead.\n";
+
+   ROOT_EXPECT_WARNING(d->AsString(), "RCsvDS", Warn);
```
Comment (eguiraud) [inline tree/dataframe/test/datasource_csv.cxx:388]: should we also test that the `Display` output contains what we expect?

**Review by eguiraud — APPROVED** [review]
Looks great, thank you very much @ikabadzhov ! :tada:

**Comment by ikabadzhov** [comment]
@phsft-bot build just on ROOT-debian10-i386/cxx14, mac1015/python3

**Comment by ikabadzhov** [comment]
I needed to rebase.

**Pull request merged**